CONTRIBUTING.md

```diff
@@ -108,6 +112,37 @@ Thank you! We'll try to respond as quickly as possible.
 
 [rustup]: https://www.rustup.rs
 
+### Coding Style
+
+We follow the [Rust Style Guide](https://github.com/rust-lang-nursery/fmt-rfcs/blob/master/guide/guide.md), enforced using [rustfmt](https://github.com/rust-lang-nursery/rustfmt).
+To run rustfmt tests locally:
+
+1. Use rustup to set rust toolchain
+   to the nightly version specified in Diesel's [.travis.yml](./.travis.yml)
+   (see above).
+
+2. Install the nightly version of rustfmt used in
+   Diesel's continuous integration.
+   Look for and run the line from [.travis.yml](./.travis.yml) that looks like
+
+   ```
+   cargo install rustfmt-nightly --vers x.x.x
+   ```
+
+3. Run rustfmt using cargo from the root of your diesel repo.
+
+   ```
+   cargo fmt --all -- --write-mode=diff
+   ```
+
+   If all code is properly formatted (e.g. if you have not made any changes),
+   this should run without error.
+   If your code needs to be reformatted,
+   you will see a diff between your code and properly formatted code.
```

Comment:
I'd tell people to just run `cargo fmt --all`, which will rewrite their code to be in Good Style. No need to read diffs :)

Reply:
My concern with that is it might be a little too magical / hidden. Maybe there's a `-v` option?

Reply:
Good point, that may throw people off when they are new to rustfmt and don't trust it yet. On the other hand… git diff before/after?

I only mention this because I have never seen anyone use rustfmt's diff thing outside of CI, but it is a valid feature :)

Reply:
@killercup I've used `write-mode=diff` recently because the local rustfmt version I have is newer than CI and caused my PR to fail after formatting. I suppose it would show up in a `git diff` anyway, but I did find it helpful.

Reply:
Is there a way to just automate this, ie have travis just run rustfmt and then add a commit fixing everything?

Reply:
Not that I'm aware of but if you know of one please do open a PR